bpf/lib/conntrack.h

```diff
@@ -557,10 +557,19 @@ static inline int __inline__ ct_create4(void *map, struct ipv4_ct_tuple *tuple,
 			cilium_trace(skb, DBG_GENERIC, ct_state->orig_dport, tuple->nexthdr);
 			/* Resolve L4 policy. This may fail due to policy reasons. May
 			 * optonally return a proxy port number to redirect all traffic to.
+			 *
+			 * However when the sender _is_ the proxy we need to ensure that
+			 * we short circuit the redirect to proxy port logic. This happens
+			 * when using ingress policies because we are doing the
+			 * l4_ingress_policy() lookup in the context of the server.
 			 */
-			proxy_port = l4_ingress_policy(skb, ct_state->orig_dport, tuple->nexthdr);
-			if (IS_ERR(proxy_port))
-				return proxy_port;
+			if (tuple->daddr == IPV4_GATEWAY) {
+				proxy_port = 0;
```

Comment:
Correct at this point the tuple has already been reversed. I have been thinking of some way to annotate this clearly in the source code but have failed so far to come up with something that does not do an equally poor job.

As for host <-> container traffic, if the IPv4 source address is the IPV4_GATEWAY address it will not be sent to the proxy in this case. I believe this is expected, do you agree?